aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -324,6 +324,9 @@ SparseTensor dense_to_sparse(const Tensor& self, int64_t sparse_dim){
 // NB: Dropped the resizeNd variants
 
 Tensor sparse_to_dense(const SparseTensor& self) {
+  if(self.scalar_type() == ScalarType::Half && !self.is_cuda()) {
```

Comment:
I think it's fine to let it error out in `add_`.  But if you want to do it this way (which improves the error message), you should check that it's on CPU, i.e. `self.device().is_cpu()` or whatever.

Reply:
ah didn't see how to get a device (via options()) and and tensor only had is_cuda